docs/framing.md

```diff
@@ -2,6 +2,32 @@
 
 Framing wraps messages with headers and checksums so receivers can identify message boundaries and verify integrity.
 
+## Frame Format Definitions
+
+All supported frame formats are defined in [`examples/frame_formats.proto`](../examples/frame_formats.proto). This file provides:
+
+- Protocol Buffer definitions for each frame format
+- Detailed documentation of use cases and tradeoffs
+- Enumeration of all frame format types
+- Configuration message for runtime format selection
+
+## Frame Format Overview
+
+| Format | Header | Footer | Length Field | Use Case |
```

Comment:
We should have recommended Frame followed by the others that are obscure variations.
The recommended one is the No Format, Basic Frame, and Basic Frame with 1 length byte.